test/Test/Toml/BiCode/Property.hs

```diff
@@ -204,3 +248,40 @@ genRec = do
     btrBoolSet <- fromList <$> Gen.list (Range.constant 0 5) genBool
     btrNewtypeList <- Gen.list (Range.constant 0 5) genSum
     pure BigTypeRecord{..}
+
+----------------------------------------------------------------------------
+-- Orphan Instances
+----------------------------------------------------------------------------
+
+instance HasCodec All where
+    hasCodec = Toml.diwrap . hasCodec @Bool
+
+instance HasCodec Any where
+    hasCodec = Toml.diwrap . hasCodec @Bool
+
+instance HasCodec a => HasCodec (Sum a) where
+    hasCodec = Toml.diwrap . hasCodec @a
+
+instance HasCodec a => HasCodec (Product a) where
+    hasCodec = Toml.diwrap . hasCodec @a
+
+instance HasCodec a => HasCodec (First a) where
+    hasCodec = Toml.diwrap . hasCodec @(Maybe a)
+
+instance HasCodec a => HasCodec (Last a) where
+    hasCodec = Toml.diwrap . hasCodec @(Maybe a)
+
+instance HasItemCodec Char where
+    hasItemCodec = Left _Char
+
+instance HasItemCodec ByteString where
+    hasItemCodec = Left Toml._ByteString
+
+instance HasCodec ByteString where
+    hasCodec = Toml.byteString
+
+instance HasCodec L.ByteString where
+    hasCodec = Toml.lazyByteString
```

Comment:
These instances are not provided by the library intentionally. I've created an issue to explain a bit motivation behind it: https://github.com/kowainik/tomland/issues/251. Could you also add a `TODO` here to remove them in the future?